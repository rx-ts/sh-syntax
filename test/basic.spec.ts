import { jest } from '@jest/globals'
import { Mock } from 'jest-mock'

import { LangVariant, parse, print } from 'sh-syntax'

const originalConsoleWarn = console.warn
let spyConsoleWarn: Mock<unknown>

beforeEach(() => {
  spyConsoleWarn = console.warn = jest.fn()
})

afterEach(() => {
  console.warn = originalConsoleWarn
})

test('it should just work', async () => {
  expect(await parse('  Hello   World!')).toMatchSnapshot()

  expect(await parse('  Hello   World ! a', { stopAt: '!' })).toMatchSnapshot()

  expect(
    await parse('  Hello   World ! b', {
      variant: LangVariant.LangPOSIX,
    }),
  ).toMatchSnapshot()

  expect(
    await parse('  Hello   World ! c', { useTabs: true }),
  ).toMatchSnapshot()

  expect(await print(null!, { filepath: 'foo.sh' })).toBe('\n')

  expect(spyConsoleWarn).toHaveBeenCalledTimes(1)

  await expect(parse('echo )')).rejects.toMatchInlineSnapshot(
    `[Error: a command can only contain words and redirects; encountered )]`,
  )
})
