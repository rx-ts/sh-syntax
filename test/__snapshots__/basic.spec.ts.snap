// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse 1`] = `
Object {
  "End": Object {
    "Col": 17,
    "Line": 1,
    "Offset": 16,
  },
  "Last": Array [],
  "Name": "",
  "Pos": Object {
    "Col": 3,
    "Line": 1,
    "Offset": 2,
  },
  "Stmt": Array [
    Object {
      "Background": false,
      "Cmd": Object {
        "End": Object {
          "Col": 17,
          "Line": 1,
          "Offset": 16,
        },
        "Pos": Object {
          "Col": 3,
          "Line": 1,
          "Offset": 2,
        },
      },
      "Comments": Array [],
      "Coprocess": false,
      "End": Object {
        "Col": 17,
        "Line": 1,
        "Offset": 16,
      },
      "Negated": false,
      "Pos": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Position": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Redirs": Array [],
      "Semicolon": Object {
        "Col": 0,
        "Line": 0,
        "Offset": 0,
      },
    },
  ],
}
`;

exports[`parse 2`] = `
Object {
  "End": Object {
    "Col": 16,
    "Line": 1,
    "Offset": 15,
  },
  "Last": Array [],
  "Name": "",
  "Pos": Object {
    "Col": 3,
    "Line": 1,
    "Offset": 2,
  },
  "Stmt": Array [
    Object {
      "Background": false,
      "Cmd": Object {
        "End": Object {
          "Col": 16,
          "Line": 1,
          "Offset": 15,
        },
        "Pos": Object {
          "Col": 3,
          "Line": 1,
          "Offset": 2,
        },
      },
      "Comments": Array [],
      "Coprocess": false,
      "End": Object {
        "Col": 16,
        "Line": 1,
        "Offset": 15,
      },
      "Negated": false,
      "Pos": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Position": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Redirs": Array [],
      "Semicolon": Object {
        "Col": 0,
        "Line": 0,
        "Offset": 0,
      },
    },
  ],
}
`;

exports[`parse 3`] = `
Object {
  "End": Object {
    "Col": 20,
    "Line": 1,
    "Offset": 19,
  },
  "Last": Array [],
  "Name": "",
  "Pos": Object {
    "Col": 3,
    "Line": 1,
    "Offset": 2,
  },
  "Stmt": Array [
    Object {
      "Background": false,
      "Cmd": Object {
        "End": Object {
          "Col": 20,
          "Line": 1,
          "Offset": 19,
        },
        "Pos": Object {
          "Col": 3,
          "Line": 1,
          "Offset": 2,
        },
      },
      "Comments": Array [],
      "Coprocess": false,
      "End": Object {
        "Col": 20,
        "Line": 1,
        "Offset": 19,
      },
      "Negated": false,
      "Pos": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Position": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Redirs": Array [],
      "Semicolon": Object {
        "Col": 0,
        "Line": 0,
        "Offset": 0,
      },
    },
  ],
}
`;

exports[`parse 4`] = `
Object {
  "End": Object {
    "Col": 20,
    "Line": 1,
    "Offset": 19,
  },
  "Last": Array [],
  "Name": "",
  "Pos": Object {
    "Col": 3,
    "Line": 1,
    "Offset": 2,
  },
  "Stmt": Array [
    Object {
      "Background": false,
      "Cmd": Object {
        "End": Object {
          "Col": 20,
          "Line": 1,
          "Offset": 19,
        },
        "Pos": Object {
          "Col": 3,
          "Line": 1,
          "Offset": 2,
        },
      },
      "Comments": Array [],
      "Coprocess": false,
      "End": Object {
        "Col": 20,
        "Line": 1,
        "Offset": 19,
      },
      "Negated": false,
      "Pos": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Position": Object {
        "Col": 3,
        "Line": 1,
        "Offset": 2,
      },
      "Redirs": Array [],
      "Semicolon": Object {
        "Col": 0,
        "Line": 0,
        "Offset": 0,
      },
    },
  ],
}
`;

exports[`print 1`] = `
"Hello World ! d
"
`;

exports[`print 2`] = `[TypeError: \`originalText\` is required for now, hope we will find better solution later]`;
